R/sass.R

```diff
@@ -55,63 +54,53 @@ sass <- function(input = NULL, options = sass_options(), output = NULL,
     stop("sass(write_attachments=TRUE) cannot be used when output=NULL")
   }
 
-  use_cache <- isTRUE(cache_options[["cache"]])
-  cache_file <- if (use_cache) {
-    cache_file_path(
-      normalizePath(cache_options[["cache_dir"]], mustWork = TRUE),
-      input,
-      options
-    )
-  }
-
   css <- NULL
   layer <- NULL
-  if (use_cache && file.exists(cache_file)) {
-    # No need to read cache file if output is specified
-    if (!is.null(output)) {
-      fs::file_copy(cache_file, output, overwrite = TRUE)
-      if (isTRUE(write_attachments == FALSE)) {
+
+  if (!is.null(cache)) {
+    cache_key <- sass_hash(list(input, options))
```

Comment:
In the event that `write_attachments = TRUE`, I think it'd make sense to `layer <- extract_layer(input)`, then include the `mtime` of any `layer$file_attachments` in the hash.

Reply:
`sass_hash()` calls `add_sass_file_mtime()` on the object passed to it. That should take care of it, right?

Reply:
No, `add_sass_file_mtime()` only works with `sass_file()`, `write_attachments` is for any files that the output css might want to import at run-time (e.g., font files)

Reply:
I'll add it, but just as a heads up, if we include the mtime of `input$file_attachments`, that will detect some kinds of changes, but not all. For example, in the case of calling `bootstrap()`, this is what it has for `input$file_attachments`: `"/Users/winston/R/4.0/bootstraplib/lib/bootstrap-sass/assets/fonts" `

By using the mtime of this directory, it will detect if any files have been added or deleted from the directory, but it won't detect if any files in the directory are modified.

Reply:
Would it make sense to do something like `file.mtime(dir(recursive = TRUE))`?